doc/manual/src/architecture/store/store.md

```diff
@@ -0,0 +1,212 @@
+# Store
+
+A Nix store is a collection of *store objects* with references between them.
+It supports operations to manipulate that collection.
+
+The following concept map is a graphical outline of this chapter.
+Arrows indicate suggested reading order.
+
+```
+                      ,--------------[ store ]----------------,
+                      |                  |                    |
+                      v                  v                    v
+               [ store object ]     [ closure ]--,      [ operations ]
+                      |               |   |      |        |        |
+                      v               |   |      v        v        |
+           [ files and processes ]    |   | [ garbage collection ] |
+               /          \           |   |                        |
+              v            v          |   v                        v
+[ file system object ] [ store path ] | [ derivation ]--->[ building ]
+                  |        ^      |   |                         |
+                  v        |      v   v                         |
+             [ digest ]----' [ reference scanning ]<------------'
+              /      \
+             v        v
+[ input addressing ] [ content addressing ]
+```
+
+## Store Object {#store-object}
+
+A store object can hold
+
+- arbitrary *data*
+- *references* to other store objects.
+
+Nix makes no distinction if store objects are build inputs, build results, or build tasks.
+
+Store objects are [immutable][immutable-object]: once created, they do not change until they are deleted.
+
+## Reference {#reference}
+
+A store object reference is an [opaque][opaque-data-type], [unique identifier][unique-identifier]:
+The only way to obtain references is by adding or building store objects.
+A reference will always point to exactly one store object.
+
```

Comment:
At this point in the narrative, store paths are a file-system-oriented implementation detail, which we try to convey in the Files and Processes section.

I know, the source code calls them store paths even if they are really store object references, given how they are used. The goal of this chapter is to get to the essence of Nix, so right now I don't care what's in the source code.

It would be great to have consistent naming. @thufschmitt agreed it would be nice in general to rename things in the source code, but it's entirely out of scope for the work here. Feel free to attack that, many people would be happy to see it happening.